c++/src/kj/async.c++

```diff
@@ -421,39 +434,99 @@ struct Executor::Impl {
     // Now we need to mark all the events "done" under lock.
     auto lock = state.lockExclusive();
     for (auto& event: eventsToCancelOutsideLock) {
-      event->state = _::XThreadEvent::DONE;
+      event->setDoneState();
     }
   }
-};
+
+  void disconnect() {
+    state.lockExclusive()->loop = nullptr;
+
+    // Now that `loop` is set null is `state`, other threads will no longer try to manipulate our
```

Comment:
Nit: "Now that `loop` is set null is `state`" sounds strange. Delete "is `state`"?

Reply:
Typo, s/is/in/